### PR TITLE
check server start() return value in stmgr unit test

### PR DESCRIPTION
**Pull request by huijunwu**
check if the server starts successfully

### CODE REVIEW

File: heron/stmgr/tests/cpp/server/stmgr_unittest.cpp [hunk 1]
```diff
@@ -228,7 +228,7 @@ void StartDummyStMgr(EventLoopImpl*& ss, DummyStMgr*& mgr, std::thread*& stmgr_t
 
   mgr = new DummyStMgr(ss, options, stmgr_id, LOCALHOST, stmgr_port, LOCALHOST, tmaster_port,
                        shell_port, instances);
-  mgr->Start();
+  EXPECT_EQ(0, mgr->Start()) << stmgr_port;
```
Comment (kramasamy) [inline heron/stmgr/tests/cpp/server/stmgr_unittest.cpp]: What does << stmgr_port means? Does it just print stmgr_port when the test fails or succeeds?

Comment (huijunw) [inline heron/stmgr/tests/cpp/server/stmgr_unittest.cpp]: When the check fails, it prints stmgr_port.
The ci seems always failing after adding the checking. I doubt if this test ever run correctly.

Comment (kramasamy) [inline heron/stmgr/tests/cpp/server/stmgr_unittest.cpp]: @huijunw - can you add some meaningful message so that it says what it is printing?

Comment (huijunw) [inline heron/stmgr/tests/cpp/server/stmgr_unittest.cpp]: done.

**Comment by kramasamy** [comment]
👍

**Pull request merged**